src/Bids/MuppetStayman.hs

```diff
@@ -0,0 +1,183 @@
+noInterference :: Action
+noInterference = P.noInterference
+b2N         :: Action
+b2N         = P.b2N
+b2N3C3D     :: Action
+b2N3C3D     = P.b2N3C3D
+b2N3C3D3H   :: Action
+b2N3C3D3H   = P.b2N3C3D3H
+b2N3C3D3H3S :: Action
+b2N3C3D3H3S = P.b2N3C3D3H3S
+b2N3C3D3H3N :: Action
+b2N3C3D3H3N = P.b2N3C3D3H3N
+b2N3C3D3S   :: Action
+b2N3C3D3S   = P.b2N3C3D3S
+b2N3C3D3S3N :: Action
+b2N3C3D3S3N = P.b2N3C3D3S3N
+b2N3C3D3S4H :: Action
+b2N3C3D3S4H = P.b2N3C3D3S4H
+b2N3C3D3N   :: Action
+b2N3C3D3N   = P.b2N3C3D3N
+b2N3C3D4D   :: Action
+b2N3C3D4D   = P.b2N3C3D4D
+b2N3C3D4D4H :: Action
+b2N3C3D4D4H = P.b2N3C3D4D4H
+b2N3C3D4D4S :: Action
+b2N3C3D4D4S = P.b2N3C3D4D4S
+b2N3C3S     :: Action
+b2N3C3S     = P.b2N3C3S
+b2N3C3S3N   :: Action
+b2N3C3S3N   = P.b2N3C3S3N
+b2N3C3S4H   :: Action
+b2N3C3S4H   = P.b2N3C3S4H
+b2N3C3S4S   :: Action
+b2N3C3S4S   = P.b2N3C3S4S
+
+
+b2N3C :: Action
+b2N3C = E.nameAction "muppet_b2N3C" $ do
+    E.pointRange 5 40  -- game forcing
+    E.alternatives [E.minSuitLength T.Hearts 3, E.minSuitLength T.Spades 3]
+    -- If you could transfer to spades, bid Muppet Stayman if you have any
+    -- interest in hearts at all. However, if you're 5-5 in the majors, transfer
+    -- to one and then bid the other yourself, rather than looking for an
+    -- unlikely 5-4 or 5-5 fit.
+    E.minSuitLength T.Spades 5 `E.impliesThat` E.minSuitLength T.Hearts 3
+    E.minSuitLength T.Spades 5 `E.impliesThat` E.maxSuitLength T.Hearts 4
+    -- If you could transfer to hearts, prefer puppet Stayman only if you've
+    -- got exactly 4 spades: with 5 or more, transfer to one and then bid the
+    -- other, and with 3 or fewer, just show the hearts.
+    E.minSuitLength T.Hearts 5 `E.impliesThat` E.suitLength T.Spades 4
+    -- Don't be interested in a minor to bid this.
+    E.forbidAll [ singleSuited T.Clubs
+                , singleSuited T.Diamonds
+                , twoSuited T.Clubs T.Diamonds
+                ]
+    -- TODO: would you bid Muppet Stayman with a void? Not sure...
+    E.makeCall $ T.Bid 3 T.Clubs
+
+
+b2N3C3H :: Action
+b2N3C3H = E.nameAction "muppet_b2N3C3H" $ do
+    E.forEach T.majorSuits (`E.maxSuitLength` 3)
+    E.makeAlertableCall (T.Bid 3 T.Hearts) "no 4-card major"
+
+
+b2N3C3H3S :: Action
+b2N3C3H3S = E.nameAction "muppet_b2N3C3H3S" $ do
+    E.maxSuitLength T.Spades 4
+    E.maxSuitLength T.Hearts 5
+    E.makeAlertableCall (T.Bid 3 T.Spades) ("relay to " .+ T.Bid 3 T.Notrump)
+
+
+b2N3C3H3S3N :: Action
+b2N3C3H3S3N = E.nameAction "muppet_b2N3C3H3S3N" $ do
+    E.makeCall $ T.Bid 3 T.Notrump
+
+
+b2N3C3H3N :: Action
+b2N3C3H3N = E.nameAction "muppet_b2N3C3H3N" $ do
+    E.suitLength T.Spades 5
+    E.makeAlertableCall (T.Bid 3 T.Notrump) "5-card spade suit, pass or correct"
+
+
+b2N3C3H3NP :: Action
+b2N3C3H3NP = E.nameAction "muppet_b2N3C3H3NP" $ do
+    E.maxSuitLength T.Spades 2
+    E.makeCall T.Pass
+
+
+b2N3C3H3N4S :: Action
+b2N3C3H3N4S = E.nameAction "muppet_b2N3C3H3N4S" $ do
+    E.minSuitLength T.Spades 3
```

Comment:
What would you do with 4333 shape with 3 spades? At matchpoints, you mightm pass 3N. Avoid the situation here.